pkg/hubble/parser/common/endpoint.go

```diff
@@ -45,23 +53,81 @@ func (r *EndpointResolver) ResolveEndpoint(ip netip.Addr, datapathSecurityIdenti
 		}
 
-		if datapathSecurityIdentity != userspaceSecurityIdentity {
-			r.log.WithFields(logrus.Fields{
-				logfields.Identity:    datapathSecurityIdentity,
-				logfields.OldIdentity: userspaceSecurityIdentity,
-				logfields.IPAddr:      ip,
-			}).Debugf("stale identity observed")
+		if datapathID != userspaceID {
+
+			localEndpoint := false
+			eps := r.endpointGetter.GetEndpoints()
+			for ep := range eps {
+				id, _ := ep.GetSecurityIdentity()
+				if id.ID == datapathID {
+					localEndpoint = true
+				}
+			}
+
+			if context.TraceObservationPoint == pb.TraceObservationPoint_TO_OVERLAY &&
+				ip == context.SrcIP && datapathID.Uint32() == context.SrcLabelID &&
+				datapathID == identity.ReservedIdentityRemoteNode &&
+				userspaceID == identity.ReservedIdentityHost {
+				// Ignore
+				//
+				// When encapsulating a packet for sending via the overlay network, if the source
+				// seclabel = HOST_ID, then we reassign seclabel with LOCAL_NODE_ID and then send
+				// a trace notify.
+			} else if context.TraceObservationPoint == pb.TraceObservationPoint_FROM_ENDPOINT &&
+				ip == context.SrcIP && datapathID.Uint32() == context.SrcLabelID &&
+				(datapathID == identity.ReservedIdentityHealth || !datapathID.IsReservedIdentity()) &&
+				userspaceID.IsWorld() {
+				// Ignore
+				//
+				// Sometimes packets from endpoint link-local addresses are intercepted by
+				// cil_from_container. Because link-local addresses are not stored in the IP cache,
+				// Hubble assigns them WORLD_ID.
+
+			} else if context.TraceObservationPoint == pb.TraceObservationPoint_FROM_HOST &&
+				ip == context.SrcIP && datapathID.Uint32() == context.SrcLabelID &&
+				datapathID.IsWorld() && userspaceID == identity.ReservedIdentityKubeAPIServer {
+				// Ignore
+				//
+				// When a pod sends a packet to the Kubernetes API, its IP is masqueraded and then
+				// when it receives a response and the masquerade is reversed, cil_from_host
+				// determines that the source ID is WORLD_ID because there is no packet mark.
+			} else if (context.TraceObservationPoint == pb.TraceObservationPoint_FROM_HOST ||
+				context.TraceObservationPoint == pb.TraceObservationPoint_TO_OVERLAY) &&
+				ip == context.SrcIP && datapathID.Uint32() == context.SrcLabelID &&
+				localEndpoint && userspaceID == identity.ReservedIdentityHost {
+				// Ignore
+				//
+				// When proxied packets (via Cilium DNS proxy) are sent from the host their source
+				// IP is that of the host, yet their security identity is retained from the
+				// original source pod.
+			} else if context.TraceObservationPoint == pb.TraceObservationPoint_TO_ENDPOINT &&
+				ip == context.SrcIP && datapathID.Uint32() == context.SrcLabelID &&
+				!datapathID.IsReservedIdentity() &&
+				(userspaceID == identity.ReservedIdentityHost || userspaceID == identity.ReservedIdentityRemoteNode) {
+				// Ignore
+				//
+				// When proxied packets (via Cilium DNS proxy) are recieved by the destination
```

Comment:
Linter found a typo here: "`recieved` is a misspelling of `received` (misspell)"